### ISSUE TITLE
Composite fields do not deserialize

### ISSUE THREAD

**Issue by doodlleus** [Innofactor/Xrm.Json.Serialization#1]
I can serialize an entity fine but it falls over when trying to deserialize if it contains a composite field. It looks like, upon serialization, it is string wrapping the value which puts in the '+' character

eg, json containing: 
"shipto_composite": "Ug1\r\nThe Rowley Building\r\n21 Queen Street\r\nLeicester\r\nLeicestershire\r\nLE1 1QD\r\nUnited Kingdom"

blows up with the error
"Error deserializing property `shipto_composite`"

**Inner exception**
"After parsing a value an unexpected character was encountered: +. Path 'shipto_composite', line 41, position 118."

"   at Newtonsoft.Json.JsonTextReader.ParsePostValue(Boolean ignoreComments)\r\n   at Newtonsoft.Json.JsonTextReader.Read()\r\n   at Innofactor.Xrm.Json.Serialization.EntityConverter.ReadJson(JsonReader reader, Type objectType, Object existingValue, JsonSerializer serializer)"

UPDATE:

Ive looked at the source and it seems to be caused by the **BasicsConverter.cs WriteJson()** function.

```
using (var converter = new StringWriter())
            {
                using (var provider = CodeDomProvider.CreateProvider("CSharp"))
                {
                    var value = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
                    provider.GenerateCodeFromExpression(new CodePrimitiveExpression(value.ToString()), converter, null);
                    var str = converter.ToString();
                }
            }
```

will output the string with it being broken in to two strings with a plus (+)

**Comment by doodlleus** [comment]
replacing the generatecodeformexpression line with the following _should_ work:
`provider.GenerateCodeFromExpression(new CodeSnippetExpression($"\"{value}\""), converter, null);`

**Comment by doodlleus** [comment]
I have created a pull request to fix the issue  (#2 ). Sorry, new to github so i may have got some  stuff wrong

**Comment by imranakram** [comment]
Thank you for the PR. It's now merged. Will try and push an updated package on nuget soon.
